### PR TITLE
added solution for homework MyLinkedList

### PR DESCRIPTION
**Pull request by Valdes2**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -3,48 +3,153 @@
...
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: mode this class at the end of the file

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -3,48 +3,153 @@
...
+        removedElement.item = null;
+        size--;
+
+        return removedItem;
     }
 
     @Override
     public T remove(T t) {
+        for (int i = 0; i < size; i++) {
+            if (t == findElement(i) || t.equals(findElement(i).item)) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]:
```suggestion
            if (t == findElement(i).item || t.equals(findElement(i).item)) {
```

Comment (Sofasmile) [reply src/main/java/core/basesyntax/MyLinkedList.java]: better create separate variable for `findElement(i).item`

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -3,48 +3,153 @@
...
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T element, Node<T> next) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: add access modifier

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 1 above)
Comment (Valdes2) [reply src/main/java/core/basesyntax/MyLinkedList.java]: fixed

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 2 above)
Comment (Valdes2) [reply src/main/java/core/basesyntax/MyLinkedList.java]: okey, fixed)

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 3 above)
Comment (Valdes2) [reply src/main/java/core/basesyntax/MyLinkedList.java]: fixed

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -93,8 +93,9 @@ public T remove(int index) {
     @Override
     public T remove(T t) {
         for (int i = 0; i < size; i++) {
-            if (t == findElement(i) || t.equals(findElement(i).item)) {
-                T removedElement = findElement(i).item;
+            Node<T> currentElement = findElement(i);
+            if (t == currentElement || t.equals(currentElement.item)) {
+                T removedElement = currentElement.item;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java]: T removedElement = currentElement.item;
                remove(i);
                return removedElement;
-->
return remove(i);

**Pull request closed without merging**